src/quic.rs

```diff
@@ -0,0 +1,66 @@
+/// This module contains optional APIs for implementing QUIC TLS.
+use client::{ClientConfig, ClientSession, ClientSessionImpl};
+use msgs::base::Payload;
+use msgs::enums::ExtensionType;
+use msgs::handshake::{ClientExtension, ServerExtension, UnknownExtension};
+use server::{ServerConfig, ServerSession, ServerSessionImpl};
+
+use std::sync::Arc;
+use webpki;
+
+/// Generic methods for QUIC sessions
+pub trait QuicExt {
```

Comment:
If this was `pub trait QuicExt: Session { ... }`, then `Box<QuicExt>` would be useful. I don't think I need this, and it's not a paradigm that composes well if we end up with various other extension traits, but it's easy.

Reply:
Actually, if we instead had `pub trait Session: QuicExt`, the composability problem is solved and I think the visibility story remains the same! Might need some gymnastics to `#[cfg]` out gracefully (say, an indirection through another trait with a blanket impl, or make the extension trait always-present but cfg its members), but should be doable.